site/app/controllers/MiscController.php

```diff
@@ -37,12 +37,19 @@ public function run() {
     }
 
     // function to check that this is a valid access request
-    private function checkValidAccess($is_zip, &$error_string, $download_with_any_role = false) {
+    private function checkValidAccess($is_zip, &$error_string, $download_with_any_role = false, $gradeable = null) {
         $error_string="";
         // only allow zip if it's a grader
         if ($is_zip) {
-            $error_string="only graders may access zip files";
-            return ($this->core->getUser()->accessGrading());
+            $error_string="only graders may access zip files of other students";
+            $can_download = (
+                $gradeable !== null
+                && !$gradeable->useVcsCheckout()
+                && $gradeable->getStudentDownload()
+                && !($gradeable->getCurrentVersionNumber() !== $gradeable->getActiveVersion() && !$gradeable->getStudentAnyVersion())
```

Comment:
Would be more clear what this is doing if it was 
` && ($gradeable->getCurrentVersionNumber() === $gradeable->getActiveVersion() || $gradeable->getStudentAnyVersion())` instead

Reply:
This logic was taken from the code that decides whether or not a version is downloadable. I'll go ahead and change that and this.